lib/components/formik/BlockNavigation.js

```diff
@@ -4,11 +4,12 @@ import { useFormikContext } from "formik";
 import { useNavPrompt } from "react-router-nav-prompt";
 import PropTypes from "prop-types";
 
-function NavigationHandler({ isDirty = false }) {
+const NavigationHandler = ({ isDirty = false }) => {
   const formikContext = useFormikContext();
   const shouldBlock =
-    isDirty || (formikContext ? Boolean(formikContext.dirty) : false);
+    isDirty || (formikContext && Boolean(formikContext.dirty));
```

Comment:
`shouldBlock` should be a boolean. if the formikContext isn't true, what will be its value?